### ISSUE TITLE
Inventory: Product Category is not translateable

### ISSUE THREAD

**Issue by PaoloJr90** [odoo/odoo#139144]
Impacted versions: 
Odoo 16 (online)

Steps to reproduce:
Attempt to enter a product category in any language

Current behavior:
There is no option to enter the translations (as with the other hierarchical elements; products, variants etc.)

Expected behavior:
Capability to enter product category translations as weith other hierarchical elements; products, variants etc.

Video/Screenshot link (optional):
please see both attached images for reference. One is for the "Product Category" name-entry; the other is for the "Product" name-entry. The "Product" does have the capability to enter it's name in other activated languages, while the product category does not.
![product](https://github.com/odoo/odoo/assets/57420955/d4afe394-db49-4ca2-bc77-addb75335f0c)
![product_category](https://github.com/odoo/odoo/assets/57420955/aa3f8a35-3d08-4209-a9f8-af5ddcb2c09f)

Support ticket number submitted via odoo.com/help (optional):

**Comment by Etren** [comment]
Indeed, this is very important for multilingual customers and teams!
I also encountered this problem.
If the product category can only be entered in a single language, the experience for multilingual users will be very bad!
When modifying the translation attributes of fields through developer mode, the system prompts that basic fields cannot be modified. Is there any other way to temporarily solve the translation problem of product categories?
